ibis/expr/selectors.py

```diff
@@ -284,3 +298,157 @@ def matches(regex: str | re.Pattern) -> Selector:
+    return functools.reduce(operator.and_, predicates)
+
+
+@public
+def c(*names: str) -> Predicate:
+    """Select specific column names."""
+    names = frozenset(names)
+    return where(lambda col: col.get_name() in names)
+
+
+class Across(Selector):
+    selector: Selector
+    funcs: Union[
+        Deferred,
+        Callable[[ir.Value], ir.Value],
+        util.frozendict[Optional[str], Union[Deferred, Callable[[ir.Value], ir.Value]]],
+    ]
+    names: Optional[Callable[[str, str], str]]
+
+    def __init__(self, selector, funcs, names) -> None:
+        super().__init__(
+            selector=selector,
+            funcs=funcs,
+            names=(
+                (lambda col, fn: col if fn is None else f"{col}_{fn}")
+                if names is None
+                else names
+            ),
+        )
+
+    def expand(self, table: ir.Table) -> Sequence[ir.Value]:
+        expanded = []
+
+        names = self.names
+        cols = self.selector.expand(table)
+        for func_name, func in self.funcs.items():
+            resolve = func.resolve if isinstance(func, Deferred) else func
+            expanded.extend(
+                col.name(names(orig_col.get_name(), func_name))
+                for col, orig_col in zip(map(resolve, cols), cols)
+            )
+
+        return expanded
+
+
+@public
+def across(selector, func, names=None) -> Across:
+    funcs = util.frozendict(func if isinstance(func, Mapping) else {None: func})
+    return Across(selector=selector, funcs=funcs, names=names)
+
+
+class IfAnyAll(Selector):
+    selector: Selector
+    predicate: Union[Deferred, Callable[[ir.Value], ir.BooleanValue]]
+    summarizer: Callable[[ir.BooleanValue, ir.BooleanValue], ir.BooleanValue]
+
+    def expand(self, table: ir.Table) -> Sequence[ir.Value]:
+        func = self.predicate
+        resolve = func.resolve if isinstance(func, Deferred) else func
+        return [
+            functools.reduce(self.summarizer, map(resolve, self.selector.expand(table)))
+        ]
+
+
+@public
+def if_any(selector: Selector, predicate: Deferred | Callable) -> IfAnyAll:
+    return IfAnyAll(selector=selector, predicate=predicate, summarizer=operator.or_)
+
+
+@public
+def if_all(selector: Selector, predicate: Deferred | Callable) -> IfAnyAll:
+    return IfAnyAll(selector=selector, predicate=predicate, summarizer=operator.and_)
+
+
+class HashableSlice(Concrete):
+    slice: slice
+
+    @classmethod
+    def __coerce__(cls, slice):
+        return cls(slice)
+
+    @property
+    def start(self):
+        return self.slice.start
+
+    @property
+    def stop(self):
+        return self.slice.stop
+
+    @property
+    def step(self):
+        return self.slice.step
+
+    @attribute.default
+    def __precomputed_hash__(self) -> int:
+        return hash((self.__class__, self.start, self.stop, self.step))
+
+
+class RangeSelector(Selector):
+    key = rlz.one_of((rlz.instance_of((str, int)), rlz.coerced_to(HashableSlice)))
```

Comment:
Type annotations here?

Reply:
How do I use `coerced_to` in a type annotation?

Reply:
Looks like maybe `Annotated` is the ticket.

Reply:
Fixed.